metadata-ingestion/src/datahub/ingestion/source/qlik_sense/qlik_sense.py

```diff
@@ -0,0 +1,529 @@
+            owner_urn=builder.make_user_urn(space.ownerId)
+            if self.config.ingest_owner and space.ownerId
+            else None,
+            created=int(space.createdAt.timestamp() * 1000),
+            last_modified=int(space.updatedAt.timestamp() * 1000),
+        )
+
+    def _gen_entity_status_aspect(self, entity_urn: str) -> MetadataWorkUnit:
+        return MetadataChangeProposalWrapper(
+            entityUrn=entity_urn, aspect=Status(removed=False)
+        ).as_workunit()
+
+    def _gen_entity_owner_aspect(
+        self, entity_urn: str, owner_id: str
+    ) -> MetadataWorkUnit:
+        aspect = OwnershipClass(
+            owners=[
+                OwnerClass(
+                    owner=builder.make_user_urn(owner_id),
+                    type=OwnershipTypeClass.DATAOWNER,
+                )
+            ]
+        )
+        return MetadataChangeProposalWrapper(
+            entityUrn=entity_urn,
+            aspect=aspect,
+        ).as_workunit()
+
+    def _gen_dashboard_urn(self, dashboard_identifier: str) -> str:
+        return builder.make_dashboard_urn(
+            platform=self.platform,
+            platform_instance=self.config.platform_instance,
+            name=dashboard_identifier,
+        )
+
+    def _gen_dashboard_info_workunit(self, sheet: Sheet) -> MetadataWorkUnit:
+        dashboard_urn = self._gen_dashboard_urn(sheet.id)
+        custom_properties: Dict[str, str] = {"chartCount": str(len(sheet.charts))}
+        dashboard_info_cls = DashboardInfoClass(
+            title=sheet.title,
+            description=sheet.description,
+            charts=[
+                builder.make_chart_urn(
+                    platform=self.platform,
+                    platform_instance=self.config.platform_instance,
+                    name=chart.qId,
+                )
+                for chart in sheet.charts
+            ],
+            lastModified=ChangeAuditStampsClass(
+                created=self._get_audit_stamp(sheet.createdAt, sheet.ownerId),
+                lastModified=self._get_audit_stamp(sheet.updatedAt, sheet.ownerId),
+            ),
+            customProperties=custom_properties,
+        )
+        return MetadataChangeProposalWrapper(
+            entityUrn=dashboard_urn, aspect=dashboard_info_cls
+        ).as_workunit()
+
+    def _gen_charts_workunit(self, charts: List[Chart]) -> Iterable[MetadataWorkUnit]:
+        """
+        Map Qlik Chart to Datahub Chart
+        """
+        for chart in charts:
+            chart_urn = builder.make_chart_urn(
+                platform=self.platform,
+                platform_instance=self.config.platform_instance,
+                name=chart.qId,
+            )
+
+            yield self._gen_entity_status_aspect(chart_urn)
+
+            yield MetadataChangeProposalWrapper(
+                entityUrn=chart_urn,
+                aspect=ChartInfoClass(
+                    title=chart.qId,
```

Comment:
these charts don't make a ton of sense - what other information can we fetch about them? at minimum, we should be able to generate a slightly better chart title and an external url pointing back to qlik

<img width="580" alt="image" src="https://github.com/datahub-project/datahub/assets/12566801/77a84936-5f07-4cfb-a457-03534aae1c5b">

Reply:
I don't see any other metadata which we can use to get proper title of chart
Below is the eg of metadata resoponse of chart which we are getting:
{
    "qInfo": {"qId": "pKtjchL", "qType": "barchart"},
    "qMeta": {"privileges": ["read"]},
    "qHyperCube": {
        "qSize": {"qcx": 3, "qcy": 9},
        "qDimensionInfo": [
            {
                "qFallbackTitle": "_FIVETRAN_SYNCED.Week",
                "qApprMaxGlyphCount": 3,
                "qCardinal": 9,
                "qSortIndicator": "A",
                "qGroupFallbackTitles": ["_FIVETRAN_SYNCED.Week"],
                "qGroupPos": 0,
                "qStateCounts": {
                    "qLocked": 0,
                    "qSelected": 0,
                    "qOption": 9,
                    "qDeselected": 0,
                    "qAlternative": 0,
                    "qExcluded": 0,
                    "qSelectedExcluded": 0,
                    "qLockedExcluded": 0,
                },
                "qTags": ["$weeknumber", "$cyclic", "$numeric", "$integer"],
                "qDimensionType": "N",
                "qGrouping": "N",
                "qNumFormat": {"qType": "U", "qnDec": 0, "qUseThou": 0},
                "qIsAutoFormat": True,
                "qGroupFieldDefs": ["_FIVETRAN_SYNCED.autoCalendar.Week"],
                "qMin": 1,
                "qMax": 53,
                "qContinuousAxes": True,
                "qIsCyclic": True,
                "qDerivedField": True,
                "qAttrExprInfo": [],
                "qAttrDimInfo": [],
                "qCardinalities": {
                    "qCardinal": 9,
                    "qHypercubeCardinal": 9,
                    "qAllValuesCardinal": -1,
                },
                "autoSort": True,
                "cId": "PGzANP",
                "othersLabel": "Others",
            },
            {
                "qFallbackTitle": "_FIVETRAN_SYNCED.Week",
                "qApprMaxGlyphCount": 3,
                "qCardinal": 9,
                "qSortIndicator": "A",
                "qGroupFallbackTitles": ["_FIVETRAN_SYNCED.Week"],
                "qGroupPos": 0,
                "qStateCounts": {
                    "qLocked": 0,
                    "qSelected": 0,
                    "qOption": 9,
                    "qDeselected": 0,
                    "qAlternative": 0,
                    "qExcluded": 0,
                    "qSelectedExcluded": 0,
                    "qLockedExcluded": 0,
                },
                "qTags": ["$numeric", "$integer"],
                "qDimensionType": "N",
                "qGrouping": "N",
                "qNumFormat": {"qType": "U", "qnDec": 0, "qUseThou": 0},
                "qIsAutoFormat": True,
                "qGroupFieldDefs": ["_FIVETRAN_SYNCED.autoCalendar.Week"],
                "qMin": 1,
                "qMax": 53,
                "qDerivedField": True,
                "qAttrExprInfo": [],
                "qAttrDimInfo": [],
                "qCardinalities": {
                    "qCardinal": 9,
                    "qHypercubeCardinal": 9,
                    "qAllValuesCardinal": -1,
                },
                "autoSort": True,
                "cId": "mpZPuW",
                "othersLabel": "Others",
            },
        ],
        "qMeasureInfo": [
            {
                "qFallbackTitle": "Sum([_FIVETRAN_SYNCED.autoCalendar.Date])",
                "qApprMaxGlyphCount": 12,
                "qCardinal": 0,
                "qSortIndicator": "D",
                "qNumFormat": {
                    "qType": "D",
                    "qnDec": 0,
                    "qUseThou": 0,
                    "qFmt": "M/D/YYYY",
                },
                "qMin": 45291,
                "qMax": 1813801836,
                "qIsAutoFormat": True,
                "qAttrExprInfo": [],
                "qAttrDimInfo": [],
                "qTrendLines": [],
                "autoSort": True,
                "cId": "jSpLmS",
                "numFormatFromTemplate": True,
            }
        ],
        "qEffectiveInterColumnSortOrder": [0, 1, 2],
        "qPivotDataPages": [],
        "qStackedDataPages": [],
        "qMode": "S",
        "qNoOfLeftDims": -1,
        "qTreeNodesOnDim": [],
        "qColumnOrder": [0, 1, 2],
        "columnOrder": [],
    },
    "script": "",
    "refLine": {"refLines": [], "dimRefLines": []},
    "showTitles": True,
    "title": "",
    "subtitle": "",
    "footnote": "",
    "disableNavMenu": False,
    "showDetails": True,
    "showDetailsExpression": False,
    "showDisclaimer": True,
    "barGrouping": {"grouping": "grouped"},
    "orientation": "vertical",
    "scrollbar": "miniChart",
    "scrollStartPos": 0,
    "gridLine": {"auto": True, "spacing": 2},
    "dataPoint": {
        "showLabels": False,
        "showSegmentLabels": False,
        "showTotalLabels": True,
    },
    "color": {
        "auto": True,
        "mode": "primary",
        "formatting": {"numFormatFromTemplate": True},
        "useBaseColors": "off",
        "paletteColor": {"index": 6},
        "useDimColVal": True,
        "useMeasureGradient": True,
        "persistent": True,
        "expressionIsColor": True,
        "expressionLabel": "",
        "measureScheme": "sg",
        "reverseScheme": False,
        "dimensionScheme": "12",
        "autoMinMax": True,
        "measureMin": 0,
        "measureMax": 10,
    },
    "legend": {"show": True, "dock": "auto", "showTitle": True},
    "dimensionAxis": {
        "continuousAuto": True,
        "show": "all",
        "label": "auto",
        "dock": "near",
        "axisDisplayMode": "auto",
        "maxVisibleItems": 10,
    },
    "preferContinuousAxis": True,
    "measureAxis": {
        "show": "all",
        "dock": "near",
        "spacing": 1,
        "autoMinMax": True,
        "minMax": "min",
        "min": 0,
        "max": 10,
    },
    "tooltip": {
        "auto": True,
        "hideBasic": False,
        "chart": {"style": {"size": "medium"}},
        "data": {},
    },
    "visualization": "barchart",
    "version": "1.34.7",
    "components": [],
    "showMiniChartForContinuousAxis": True,
}

Reply:
If the object has a title set in Qlik, we should figure out how to extract that. Otherwise, let's set the names to "Object 1", "Object 2" etc

Also, the dimensions and measures should get extracted into the `InputFieldsClass` aspect on the chart